### PR TITLE
Fix add-to-waitlist dropdown menu

### PR DESCRIPTION
**Pull request by candela97**
Fix dropdown display due to Steam changes described in #1085.

### CODE REVIEW

File: src/js/Content/Features/Store/App/FWaitlistDropdown.js [hunk 1]
```diff
@@ -5,67 +5,56 @@ import {Page} from "../../Page";
 export default class FWaitlistDropdown extends Feature {
 
     checkPrerequisites() {
-        return document.querySelector("#add_to_wishlist_area")
-            && SyncedStorage.get("add_to_waitlist")
-            && Background.action("itad.isconnected");
+        return SyncedStorage.get("add_to_waitlist")
+            && Background.action("itad.isconnected")
+            && document.querySelector("#add_to_wishlist_area") !== null;
```
Comment (candela97) [inline src/js/Content/Features/Store/App/FWaitlistDropdown.js:10]: This basically checks if the app is owned or not now, maybe our `isOwned()` can be replaced with this.

File: src/js/Content/Features/Store/App/FWaitlistDropdown.js [hunk 2]
```diff
@@ -5,67 +5,56 @@ import {Page} from "../../Page";
...
             </div>`);
 
         /*
          * Creating a common parent for #add_to_wishlist_area and #add_to_wishlist_area_success
          * makes it easier to apply the dropdown menu
          */
-        const wrapper = document.querySelector(".queue_btn_wishlist");
-
-        // Move the wrapper such that there can't be any other elements in between the dropdown and other buttons (see #690)
-        document.querySelector(".queue_actions_ctn").insertBefore(wrapper.parentNode, wishlistArea);
```
Comment (candela97) [inline src/js/Content/Features/Store/App/FWaitlistDropdown.js:37]: This workaround shouldn't be needed anymore.

**Comment by candela97** [comment]
Should we add back the "Manage your wishlist" link?
螢幕擷取畫面 2021-02-27 175615

**Comment by Revadike** [comment]
Instead of removing the dropdown box altogether, perhaps _augment_ it instead. I guess both options are equally fine, but I think building on top of steam's functionality is probably better than to replace it. What do you think?

**Comment by candela97** [comment]
Seems like the waitlist menu was designed to mimic the ignore menu (which has better UX if there're multiple options), and also aims to have the same menu for both wishlist buttons (as opposed to Steam's which only appears on the wishlist success button), so it's easier to work with if we just replace the whole thing...

File: src/js/Content/Features/Store/App/FWaitlistDropdown.js (hunk 1 above)
Comment (MxtOUT) [reply src/js/Content/Features/Store/App/FWaitlistDropdown.js:10]: The first condition is not related to ownership, the second only works when you're connected to ITAD and the third is already included in `isOwned`.
I think we should distinguish between "owned" (Steam) and "collected" (ITAD)

**Review by MxtOUT — APPROVED** [review]
To me this new dropdown menu looks like they tried to find a compromise between their and the user's interests.
For monetary reasons they still hold on to the useless wishlist link (it's always in the top right corner and why would you want to visit your wishlist after wishlisting an item?) and miss the opportunity to properly implement this feature.
Notice how you have two options for viewing your wishlist now.

Anyway, I think replacing it is the way to go.
How would you otherwise integrate this into a "Ignore Menu"-fashioned dropdown?
Also add a link to the user's Waitlist? I don't think so

**Pull request merged**